utils/display.go

```diff
@@ -873,53 +872,92 @@ func AppTable(app *meroxa.Application) string {
 		})
 	}
 
-	if len(app.Connectors) != 0 {
-		names := make([]string, 0)
-		for _, f := range app.Connectors {
-			id, err := f.GetNameOrUUID()
-			if err != nil {
-				continue
-			}
-			names = append(names, id)
-		}
+	mainTable.SetStyle(simpletable.StyleCompact)
+	return mainTable.String()
+}
 
-		mainTable.Body.Cells = append(mainTable.Body.Cells, []*simpletable.Cell{
-			{Align: simpletable.AlignRight, Text: "Connectors:"},
-			{Text: strings.Join(names, ", ")},
-		})
+func ExtendedAppTable(app *meroxa.Application, resources []*meroxa.Resource, connectors map[string]*meroxa.Connector,
+	functions []*meroxa.Function) string {
+	mainTable := simpletable.New()
+	mainTable.Body.Cells = [][]*simpletable.Cell{
+		{
+			{Align: simpletable.AlignRight, Text: "UUID:"},
+			{Text: app.UUID},
+		},
+		{
+			{Align: simpletable.AlignRight, Text: "Name:"},
+			{Text: app.Name},
+		},
+		{
+			{Align: simpletable.AlignRight, Text: "Language:"},
+			{Text: app.Language},
+		},
+		{
+			{Align: simpletable.AlignRight, Text: "Git SHA:"},
+			{Text: strings.TrimSpace(app.GitSha)},
+		},
+		{
+			{Align: simpletable.AlignRight, Text: "Created At:"},
+			{Text: app.CreatedAt.String()},
+		},
+		{
+			{Align: simpletable.AlignRight, Text: "Updated At:"},
+			{Text: app.UpdatedAt.String()},
+		},
+		{
+			{Align: simpletable.AlignRight, Text: "State:"},
+			{Text: strings.Title(string(app.Status.State))},
+		},
 	}
-	if len(app.Functions) != 0 {
-		names := make([]string, 0)
-		for _, f := range app.Functions {
-			id, err := f.GetNameOrUUID()
-			if err != nil {
-				continue
-			}
-			names = append(names, id)
-		}
+	mainTable.SetStyle(simpletable.StyleCompact)
+	output := mainTable.String()
 
-		mainTable.Body.Cells = append(mainTable.Body.Cells, []*simpletable.Cell{
-			{Align: simpletable.AlignRight, Text: "Functions:"},
-			{Text: strings.Join(names, ", ")},
-		})
+	subTable := extendedResourcesTable(resources, connectors)
+	if subTable != "" {
+		output += "\n" + subTable
 	}
-	if len(app.Resources) != 0 {
-		names := make([]string, 0)
-		for _, f := range app.Resources {
-			id, err := f.GetNameOrUUID()
-			if err != nil {
-				continue
-			}
-			names = append(names, id)
+	subTable = extendedFunctionsTable(functions)
+	if subTable != "" {
+		output += "\n" + subTable
+	}
+	return output
+}
+
+func extendedResourcesTable(resources []*meroxa.Resource, connectors map[string]*meroxa.Connector) string {
```

Comment:
i'm still not sure why, but the typical library used for these tables was giving me lots of issues with nil pointers when trying to format these two subtables, so i just went with a less flexible option